src/Extensions/Abstractions/IMachineInformation.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System.Net;
+
+namespace Microsoft.Omex.Extensions.Abstractions
+{
+	/// <summary>Machine information interface</summary>
+	public interface IMachineInformation
```

Comment:
- BuildVersion
- DeploymentSlice
- IsPrivateDeployment
- ServiceName

are deployment related information and not necessarily machine per se. I think we should extract these to their own interface `IDeploymentInformation` or `IBuildInformation` I think `AgentName` should be deprecated since we have the `ServiceName` to replace that information now.

Reply:
I totally agree with this, but If you don't mind I would like to do it in a separate PR since it's spaning some discussions. I'll try to minimize the usage of this interface for now, sorry

Reply:
I've played with this idea for some time since I've got more similar comments.
For some of the properties, it's a bit complicated to draw the border (ex. Environment).
Also, spliting them complicated because most of information coming from the same sources and have similar usage. 
So, for now, I've updated names for some properties and renamed interface to `IExecutionContext` since it's providing information about the environment where code is executing.